control-plane/agents/src/bin/core/volume/operations.rs

```diff
@@ -185,10 +185,41 @@ impl ResourceResize for OperationGuardArc<VolumeSpec> {
 
     async fn resize(
         &mut self,
-        _registry: &Registry,
-        _request: &Self::Resize,
+        registry: &Registry,
+        request: &Self::Resize,
     ) -> Result<Self::ResizeOutput, SvcError> {
-        unimplemented!()
+        let specs = registry.specs();
+        let spec = self.as_ref().clone();
+        let state = registry.volume_state(&request.uuid).await?;
+        let replicas = specs.volume_replicas(&request.uuid);
+
+        // Pre-Check - volume state ok for resize.
+        if !spec.vol_status_ok_for_resize() {
+            return Err(SvcError::InUse {
+                kind: stor_port::transport_api::ResourceKind::Volume,
+                id: spec.uuid.to_string(),
+            });
+        }
+
+        let spec_clone = self
+            .start_update(
+                registry,
+                &state,
+                VolumeOperation::Resize(request.requested_size),
+            )
+            .await?;
+        // For the volume being resized, resize each replica of the volume. For a failure to
+        // resize any of the replica, the volume resize operation is to be deemed failure.
+        match self
+            .resize_volume_replicas(registry, &replicas, request.requested_size, &spec_clone)
+            .await
+        {
+            Ok(_) => {
+                let volume = registry.volume(&request.uuid).await?;
+                self.complete_update(registry, Ok(volume), spec_clone).await
+            }
+            Err(e) => Err(e),
```

Comment:
Can we not pass the result directly to the complete_update?

Reply:
> Can we not pass the result directly to the complete_update?

We can if we make `resize_volume_replicas` return a `Volume` object. Keeping it separate would be required in the case when we have to work on nexus after getting Ok() here.

Reply:
We cannot return error from line 218, it's again the same issue.
Also we cannot simply return from 221 since update has started.

So rather we can do this ?

```suggestion
           let result = match self
            .resize_volume_replicas(registry, &replicas, request.requested_size, &spec_clone)
            .await
        {
            Ok(_) => {
                match registry.volume(&request.uuid).await {
                   Ok(volume) => Ok(volume),
                   Err(e) => Err(e)
            }
            Err(e) => Err(e),
        }
         self.complete_update(registry, result, spec_clone).await
```

Reply:
ooh yike, I missed that line #218 call. #221 would've been ok I think because before that `validate` has been done internally by helper function. I'll fix based on suggestion.

Reply:
Modified.